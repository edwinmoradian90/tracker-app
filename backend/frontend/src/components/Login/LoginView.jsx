import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { GeneralButton, Input, Error, fadein } from '../../utils/styles/generalStyles';
import { green, medGrey, white, warningRed, line } from '../../utils/colors/main';

const LoginContainer = styled.div`
    display: flex;
    flex-direction: column;
    padding: 20px;
`;

const LoginTitle = styled.div`
    margin: 20px 0;
`;

const NewUser = styled.p`
    animation: ${fadein} 1.5s 1;
    color: ${medGrey};
    opacity: .7;
    margin: 10px auto;
`;

const InputContainer = styled.div`
    margin-bottom: 0;
    width: 100%;
`;

const LoginView = props => {
    const { onChange, onSubmit, state, incorrectData } = props;
    const { email, password } = state;
    const validations = email && password && password.length >= 6;
    return (
        <LoginContainer className="loginView">
            <LoginTitle className="loginTitle">Please login or sign up to continue.</LoginTitle>
            <InputContainer className="loginInputContainer">
                <Input
                    marginBottom="20px"
                    onChange={onChange}
                    name="email"
                    type="email"
                    className="email"
                    placeholder="Email"
                    value={email}
                    width="100%"
                />
                <Input
                    onChange={onChange}
                    name="password"
                    type="password"
                    className="password"
                    placeholder="Password"
                    value={password}
                    width="100%"
                />
            </InputContainer>
            <Error
                margin="20px 20px 20px 0px"
                incorrectData={incorrectData}
            >
                Please check your email or password.
            </Error>
            <GeneralButton
                animation={true}
                background={validations ? green : medGrey}
                color={white}
                className="submitLoginForm"
                onClick={
                    validations
                        ?
                        e => onSubmit(e)
                        :
                        null
                }
                type="submit"
            >
                Log in
            </GeneralButton>
            <NewUser className="newUserSignupLink">
                New user? {" "}
                <Link to="/signup">
                    Sign up.
                </Link>
            </NewUser>
        </LoginContainer>
    );
};

const { func, object, bool } = PropTypes;
LoginView.propTypes = {
    onChange: func.isRequired,
    onSubmit: func.isRequired,
    state: object.isRequired,
    incorrectData: bool.isRequired,
};

export default LoginView;