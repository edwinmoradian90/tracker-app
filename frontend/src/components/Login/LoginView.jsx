import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { GeneralButton, fadein } from '../../utils/styles/generalStyles';
import { green, medGrey, white } from '../../utils/colors/main';

const LoginContainer = styled.div`
    display: flex;
    flex-direction: column;
    padding: 20px;
`;

const LoginTitle = styled.div`
    margin: 20px 0;
`;

const NewUser = styled.p`
    animation: ${fadein} 1.5s 1;
    color: ${medGrey};
    opacity: .7;
    margin: 10px auto;
`;

const InputContainer = styled.div`
    margin-bottom: 0;
`;

const Input = styled.input`
    margin-bottom: ${props => props.marginBottom};
    width: 94%;
`;

const LoginView = props => {
    const { onChange, onSubmit, state } = props;
    const { email, password } = state;
    return (
        <LoginContainer className="loginView">
            <LoginTitle className="loginTitle">Please login or sign up to continue.</LoginTitle>
            <div className="loginInputContainer">
                <InputContainer className="emailInputContainer">
                    <Input
                        marginBottom="20px"
                        onChange={onChange}
                        name="email"
                        type="email"
                        className="email"
                        placeholder="Email"
                        value={email}
                    />
                </InputContainer>
                <InputContainer className="passwordInputContainer">
                    <Input
                        onChange={onChange}
                        name="password"
                        type="password"
                        className="password"
                        placeholder="Password"
                        marginBottom="20px"
                        value={password}
                    />
                </InputContainer>
            </div>
            <GeneralButton
                animation={true}
                background={email && password ? green : medGrey}
                color={white}
                className="submitLoginForm"
                onClick={
                    email && password
                        ?
                        e => onSubmit(e)
                        :
                        null
                }
                type="submit"
            >
                Log in
            </GeneralButton>
            <NewUser className="newUserSignupLink">
                New user? {" "}
                <Link to="/signup">
                    Sign up.
                </Link>
            </NewUser>
        </LoginContainer>
    );
};

const { func, object } = PropTypes;
LoginView.propTypes = {
    onChange: func.isRequired,
    onSubmit: func.isRequired,
    state: object.isRequired,
};

export default LoginView;