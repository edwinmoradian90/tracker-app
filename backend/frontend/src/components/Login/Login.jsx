import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { currentUser } from '../../redux/actions/sessions';
import LoginView from './LoginView';
import axios from 'axios';

const Login = props => {
    const dispatch = useDispatch();
    const url = "/login";
    const [state, setState] = useState({
        email: "",
        password: "",
    });
    const [incorrectData, setIncorrectData] = useState(false);
    const [submitForm, setSubmitForm] = useState(false);
    const currentUserCheck = () => {
        return localStorage.getItem('currentUser');
    };
    const removeCurrentUser = () => {
        localStorage.removeItem('currentUser');
    };
    const onChange = e => {
        if (incorrectData) {
            setIncorrectData(false);
        };
        if (submitForm) {
            setSubmitForm(false);
        };
        setState({
            ...state,
            [e.target.name]: e.target.value
        });
    };
    const onSubmit = e => {
        e.preventDefault();
        if (!submitForm) {
            setSubmitForm(true);
        };
        if (currentUserCheck()) {
            removeCurrentUser();
        };
        const { email, password } = state;
        const user = { email, password };
        axios.post(url, user)
            .then(res => {
                const { status } = res.data;
                if (status !== 200) {
                    setIncorrectData(true);
                } else if (status === 200) {
                    const { data } = res;
                    const token = data.jwt;
                    const {
                        first_name,
                        last_name,
                        email,
                        updated_at,
                        created_at,
                        id,
                    } = data.user;
                    const userInfo = {
                        firstName: first_name,
                        lastName: last_name,
                        email: email,
                        updatedAt: updated_at,
                        createdAt: created_at,
                        id: id,
                        token: `Bearer ${token}`,
                    };
                    if (token) {
                        const newUrl = "/user_is_authed";
                        const headers = {
                            "Authorization": `Bearer ${token}`,
                        };
                        axios.get(newUrl, { headers })
                            .then(res => {
                                const { status } = res.data;
                                localStorage.setItem('currentUser', JSON.stringify(userInfo));
                                dispatch(currentUser(userInfo));
                                props.history.push('/');
                            })
                            .catch(err => console.log(err));
                    };
                }
            })
            .catch(err => console.log(err));
    }
    return (
        <LoginView
            onChange={onChange}
            onSubmit={onSubmit}
            submitForm={submitForm}
            state={state}
            incorrectData={incorrectData}
        />
    );
};

export default withRouter(Login);