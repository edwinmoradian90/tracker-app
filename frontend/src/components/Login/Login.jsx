import React, { useState, useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { currentUser } from '../../redux/actions/sessions';
import LoginView from './LoginView';
import axios from 'axios';

const Login = props => {
    const dispatch = useDispatch();
    const url = "http://localhost:3001/login"
    const [state, setState] = useState({
        email: "",
        password: "",
    });
    const onChange = e => {
        setState({
            ...state,
            [e.target.name]: e.target.value
        });
        console.log(state);
    };
    const onSubmit = e => {
        e.preventDefault();
        const { email, password } = state;
        const user = { email, password };
        axios.post(url, user)
            .then(res => {
                const { data } = res;
                console.log(data)
                const token = data.jwt;
                const {
                    first_name,
                    last_name,
                    email,
                    updated_at,
                    created_at,
                    id,
                } = data.user;
                const userInfo = {
                    firstName: first_name,
                    lastName: last_name,
                    email: email,
                    updatedAt: updated_at,
                    createdAt: created_at,
                    id: id,
                    token: `Bearer ${token}`,
                };
                if (token) {
                    console.log('there is a token');
                    const newUrl = "http://localhost:3001/user_is_authed";
                    const headers = {
                        "Authorization": `Bearer ${token}`,
                    };
                    axios.get(newUrl, { headers })
                        .then(res => console.log('headers', res))
                        .catch(err => console.log(err));
                    localStorage.setItem('currentUser', JSON.stringify(userInfo));
                    dispatch(currentUser(userInfo));
                    props.history.push('/');
                };
            })
            .catch(err => console.log(err));
    }
    return (
        <LoginView
            onChange={onChange}
            onSubmit={onSubmit}
        />
    );
};

export default withRouter(Login);