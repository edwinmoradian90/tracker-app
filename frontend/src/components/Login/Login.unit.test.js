import React from 'react';
import renderer from 'react-test-renderer';
import Login from './Login';

describe('Login', () => {
    it('Renders', () => {
        const component = renderer.create(
            <Login />
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});