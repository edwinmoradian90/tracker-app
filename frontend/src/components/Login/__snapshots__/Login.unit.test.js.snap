// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Renders 1`] = `
<div
  className="sc-gtssRu jPJwkn loginView"
>
  <div
    className="sc-dlnjPT dpOVoQ loginTitle"
  >
    Please login or sign up to continue.
  </div>
  <div
    className="loginInputContainer"
  >
    <div
      className="sc-eCApGN dCxyJq emailInputContainer"
    >
      <input
        className="sc-jSFkmK ebVQMe email"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        type="email"
        value=""
      />
    </div>
    <div
      className="sc-eCApGN dCxyJq passwordInputContainer"
    >
      <input
        className="sc-jSFkmK ebVQMe password"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
  </div>
  <div
    className="sc-bdnylx jCKEPf submitLoginForm"
    color="#FFFFFF"
    onClick={null}
    type="submit"
  >
    Log in
  </div>
  <p
    className="sc-hKFyIo gOCaNB newUserSignupLink"
  >
    New user? 
     
    <a
      href="/signup"
      onClick={[Function]}
    >
      Sign up.
    </a>
  </p>
</div>
`;
